src/main/java/org/openrewrite/java/testing/junit5/MigrateFromHamcrest.java

```diff
@@ -0,0 +1,119 @@
+                if (hamcrestMatcher instanceof J.MethodInvocation) {
+                    J.MethodInvocation matcherInvocation = (J.MethodInvocation)hamcrestMatcher;
+                    maybeRemoveImport("org.hamcrest.Matchers." + matcherInvocation.getSimpleName());
+                    maybeRemoveImport("org.hamcrest.MatcherAssert.assertThat");
+                    String targetAssertion = getTranslatedAssert(matcherInvocation);
+                    JavaTemplate template = JavaTemplate.builder(this::getCursor, getTemplateForTranslatedAssertion(targetAssertion))
+                      .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "junit-jupiter-api-5.9"))
+                      .staticImports("org.junit.jupiter.api.Assertions." + targetAssertion)
+                      .build();
+                    mi = withTemplate(mi, template, mi.getArguments().get(0), stripMatcherInvocation(mi.getArguments().get(1)));
+                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
+                }
+                else throw new IllegalArgumentException("Parameter mismatch for " + mi + ".");
+            }
+            return mi;
+        }
+
+        private J.MethodInvocation withTemplate(J.MethodInvocation method, JavaTemplate template, Expression firstArg, List<Expression> matcherArgs) {
+            switch (matcherArgs.size()) {
+                case 0:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg);
+                case 1:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0));
+                case 2 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1));
+                case 3 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1), matcherArgs.get(2));
+                default:
+                    throw new IllegalArgumentException("List of matcher arguments is too long for " + method + ".");
+            }
+        }
+
+        private List<Expression> stripMatcherInvocation(Expression e) {
+            if (e instanceof J.MethodInvocation) {
+                MethodMatcher matchesMatcher = new MethodMatcher("org.hamcrest.Matchers *(..)");
+                if (matchesMatcher.matches(e)) {
+                    return ((J.MethodInvocation) e).getArguments();
+                }
+            }
+            throw new IllegalArgumentException("Trying to strip an expression which is not a matcher invocation:\n" + e);
+        }
+
+        private String getTranslatedAssert(J.MethodInvocation methodInvocation) {
+            //to be replaced with a static map
+            switch (methodInvocation.getSimpleName()) {
+                case "equalTo":
+                    return "assertEquals";
```

Comment:
Rather than statically defining the various options here, we could also look at turning the `hamcrestMatcher` and `targetAssertion` into `@Option` arguments to the recipe, such that we can configure this repeatedly from a declarative Yaml description file. We already do [something similar for Mockito](https://github.com/openrewrite/rewrite-testing-frameworks/blob/main/src/main/resources/META-INF/rewrite/mockito.yml), except in thise case you'd be configuring this new recipe rather than `ChangeMethodName`. Such an approach pull those replacements out of code and into configuration, which is typically easier to expand on, even locally, for our users.